inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/plugin/PluginService.java

```diff
@@ -50,7 +50,7 @@
     private List<PluginBinder> pluginBinders;
 
     @Getter
-    private List<Plugin> plugins = new ArrayList<>();
+    private final List<Plugin> plugins = new ArrayList<>();
```

Comment:
for uniqueness, maybe using a Set<Plugin> here is better

Reply:
Each plugin instance was created by `new`, and not overwrite the `toString` and `equals` methods, so `Set` and `List` are the same in this scenario.

Reply:
@kipshi Could you confirm that all the plugins here should be a singleton?
If so, current logic may break this semantic, e.g. if we have two `PluginDefinitions` with the same `pluginClass` and different `name`, the `List<Plugin> plugins` will have two instances for the same `pluginClass`.